cmd/ingress-operator/start.go

```diff
@@ -123,7 +124,15 @@ func start(opts *StartOptions) error {
 	}
 
 	// Start operator metrics.
-	go canarycontroller.StartMetricsListener(opts.MetricsListenAddr, signal)
+	go operator.StartMetricsListener(opts.MetricsListenAddr, signal)
+	log.Info("registering Prometheus metrics for canary_controller")
+	if err := canarycontroller.RegisterMetrics(); err != nil {
+		log.Error(err, "unable to register metrics")
```

Comment:
Would be useful to indicate which registration failed as we have two cases.

Reply:
Done.